lib/matplotlib/text.py

```diff
@@ -1380,6 +1385,32 @@ def set_fontname(self, fontname):
         """
         self.set_fontfamily(fontname)
 
+    def _ha_for_angle(self, angle):
+        """
+        Determines horizontal alignment ('ha') for rotation_mode "xtick" based on
+        the angle of rotation in degrees and the vertical alignment.
+        """
+        anchor_at_bottom = self.get_verticalalignment() == 'bottom'
+        if (angle < 5 or 85 <= angle < 105 or 355 <= angle < 360 or
+                170 <= angle < 190 or 265 <= angle < 275):
+            return 'center'
+        elif 5 <= angle < 85 or 190 <= angle < 265:
+            return 'left' if anchor_at_bottom else 'right'
+        return 'right' if anchor_at_bottom else 'left'
+
+    def _va_for_angle(self, angle):
+        """
+        Determines vertical alignment ('va') based on the angle of rotation
+        in degrees. Adjusts for is_tick_right_enabled.
```

Comment:
like for `_ha_for_angle`:
```suggestion
        Determines vertical alignment ('va') for rotation_mode "ytick" base on
        the angle of rotation in degrees and the horizontal alignment.
```